common/test/acceptance/pages/lms/instructor_dashboard.py

```diff
@@ -685,12 +685,47 @@ class DataDownloadPage(PageObject):
     def is_browser_on_page(self):
         return self.q(css='a[data-section=data_download].active-section').present
 
+    @property
+    def generate_student_profile_report_button(self):
+        """
+        Returns the "Download profile information as a CSV" button.
+        """
+        return self.q(css='input[name=list-profiles-csv]')
+
+    @property
+    def generate_grade_report_button(self):
+        """
+        Returns the "Generate Grade Report" button.
+        """
+        return self.q(css='input[name=calculate-grades-csv]')
+
+    @property
+    def generate_weighted_problem_grade_report_button(self):
+        """
+        Returns the "Generate Weighted Problem Grade Report" button.
+        """
+        return self.q(css='input[name=problem-grade-report]')
+
+    @property
+    def report_download_links(self):
+        """
+        Returns the download links for the current page.
+        """
+        return self.q(css="#report-downloads-table .file-download-link>a")
+
+    def wait_for_available_report(self):
+        """
+        Waits for a downloadable report to be available.
+        """
+        EmptyPromise(
+            lambda: len(self.report_download_links) >= 1, 'Waiting for downloadable report'
```

Comment:
I was following the approach we took for the learner profile events where we capture whatever is there. The benefit to this approach is if for whatever reason there are two reports then the promise will still return. If we wait for exactly one report then the promise will time out with an unclear failure message that will make it seem that no report was generated.

Having said this, the learner profile event tests relied upon the tests themselves doing a diff on the whole list, so it was clear what the failure was. I think I should update my test to do the same thing, rather than just pull out the first report.

Note that it is very unlikely that there's another report as each test gets a unique course. It was much more of an issue for events where all sorts of events could be fired during a test run. However, it is better to be very clear in the tests.